### PR TITLE
implementing GONE pipeline

### PR DESCRIPTION
**Pull request by chriscrsmith**
Added snakemake rules for running GONE pipeline for estimating Ne. A few things to note:

- So far I haven't done anything to tease out neutral mutations (as in #44 )
- Some of the c executables for GONE don't let you redirect input/output. So I included some shenanigans for editing the GONE params files and copying them into local working directories. Another option is to edit the source code. Open to suggestions.

### CODE REVIEW

File: workflows/config/snakemake/config.yaml [hunk 1]
```diff
@@ -32,3 +32,4 @@
 "poly_dfe_exec": "ext/polyDFE/polyDFE-2.0-linux-64-bit"
 "msmc_exec" : "ext/msmc/build/msmc"
 "stairwayplot_code" : "stairwayplot/swarmops.jar"
+"gone_code" : "GONE/Linux"
```
Comment (izabelcavassim) [inline workflows/config/snakemake/config.yaml:35]: does it only work with linux machine?

File: workflows/gone.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    # subprocess.run(cmd, shell=True, check=True)
+
+
+    
+def params(gone_code):
+    """
+    modifies the GONE params file
+    """
+    phase=1 # 0 for pseudohaploid, 1 for unknown phase, 2 for phased
+    max_snps=500 # default=50000
```
Comment (izabelcavassim) [inline workflows/gone.py:21]: can't we have variables specifying this?

File: workflows/config/snakemake/config.yaml (hunk 1 above)
Comment (chriscrsmith) [reply workflows/config/snakemake/config.yaml:35]: linux or macos. Think that should suffice?

File: workflows/gone.py (hunk 2 above)
Comment (chriscrsmith) [reply workflows/gone.py:21]: Good idea you mean add these to the config file?

Comment (andrewkern) [reply workflows/gone.py:21]: sounds like a good plan to put this in the config file

**Comment by chriscrsmith** [comment]
> @chriscrsmith will this run if I pull the PR and try to run the whole pipeline? i may have time today to play around

yes it should

**Comment by andrewkern** [comment]
tried to run this and got the following error

```
RuleException in line 370 of /home/adkern/popSim/analysis2/workflows/n_t.snake:
NameError: The name 'gone_code' is unknown in this context. Please make sure that you defined that variable. Also note that braces not used for variable access have to be escaped by repeating them, i.e. {{print $1}}
ERROR:snakemake.logging:RuleException in line 370 of /home/adkern/popSim/analysis2/workflows/n_t.snake:
NameError: The name 'gone_code' is unknown in this context. Please make sure that you defined that variable. Also note that braces not used for variable access have to be escaped by repeating them, i.e. {{print $1}}
```

**Comment by chriscrsmith** [comment]
> tried to run this and got the following error
> 
> ```
> RuleException in line 370 of /home/adkern/popSim/analysis2/workflows/n_t.snake:
> NameError: The name 'gone_code' is unknown in this context. Please make sure that you defined that variable. Also note that braces not used for variable access have to be escaped by repeating them, i.e. {{print $1}}
> ERROR:snakemake.logging:RuleException in line 370 of /home/adkern/popSim/analysis2/workflows/n_t.snake:
> NameError: The name 'gone_code' is unknown in this context. Please make sure that you defined that variable. Also note that braces not used for variable access have to be escaped by repeating them, i.e. {{print $1}}
> ```

I think this is a bug with snakemake. The docs say to use python variables inside shell commands with single braces, double braces are for escaping the inner braces so that they make it to the command line.

I'm sure it's possible to use different syntax with the shell commands but I thought I'd try skipping all of that, just moved the shell commands into python. Ugly. But it's working for me. I ran the whole pipeline successfully with `snakemake -c 20`

**Comment by andrewkern** [comment]
nice @chriscrsmith this working for me.

**Review by andrewkern — APPROVED** [review]

**Comment by andrewkern** [comment]
this is looking good to me and ready to merge. any reason I should hold off here @chriscrsmith?

**Comment by chriscrsmith** [comment]
I probably need to squash+rebase yeah?

**Comment by andrewkern** [comment]
yes plz

**Pull request merged**